script.js

```diff
@@ -225,101 +231,165 @@ const generateBoard = () => {
 // Randomly select a person from the characters array and set as the value of the variable called secret
 const setSecret = () => {
   secret = charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)]
-}
+  console.log('secret')
+};
 
 // This function to start (and restart) the game
 const start = () => {
-  // Here we're setting charactersInPlay array to be all the characters to start with
+  numberOfGuesses = 0
   charactersInPlay = CHARACTERS
-  // What else should happen when we start the game?
-}
+  generateBoard()
+  setSecret()
+  guessCounter.innerHTML = `<p>Number of guesses: ${numberOfGuesses}/4</p>`
+  console.log('start')
+};
 
 // setting the currentQuestion object when you select something in the dropdown
+// bot is playing with user.
 const selectQuestion = () => {
   const category = questions.options[questions.selectedIndex].parentNode.label
-
+  const attribute = questions.options[questions.selectedIndex].parentNode.label 
   // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
-
+  const value = questions.value
+  // This variable stores the actual value of the question we've selected.
   currentQuestion = {
     category: category,
-    // value: value
+    attribute: attribute,
+    value: value
   }
-}
+
+  console.log('select-question')
+};
 
 // This function should be invoked when you click on 'Find Out' button.
+// Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
+// See if we should keep or remove people based on that
 const checkQuestion = () => {
-  const { category, value } = currentQuestion
+  let keep
+  const { category, value, attribute } = currentQuestion
+  if (attribute === 'hair' || attribute === 'eyes') {
+    value === secret[category]
+  }
+  else if (category === 'accessories' || category === 'other') {
+    value === secret[category].includes(value)
+  }
 
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
-  // See if we should keep or remove people based on that
   // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
-
-  } else if (category === 'accessories' || category === 'other') {
-
+  filterCharacters(keep)
+  numberOfGuesses += 1
+  if (numberOfGuesses === 4) {
+    alert('Your are out of guesses. Please make your final choice.')
+    filter.disabled = true
```

Comment:
I liked that you dont loose (game over) when you are out of guesses and that you are able to select a guess.